ui/pages/permissions-connect/connect-page/utils.ts

```diff
@@ -101,6 +108,21 @@ export function getDefaultAccounts(
     }
   });
 
+  if (
+    existingPermittedCaipAccountIds &&
+    existingPermittedCaipAccountIds.length > 0
+  ) {
+    const addresses = existingPermittedCaipAccountIds.map((caipAccountId) => {
+      const { address } = parseCaipAccountId(caipAccountId);
+      return address.toLowerCase();
+    });
+
+    const accounts = allAccounts.filter((account) => {
+      return addresses.includes(account.address.toLowerCase());
+    });
+    defaultAccounts.push(...accounts);
```

Comment:
if our existing permitted account ids did not exist in all accounts, then something is very wrong. Is this check needed? Even better yet, does `existingPermittedCaipAccountIds` need to be inside getDefaultAccounts at all or can it be appended outside?

Reply:
> if our existing permitted account ids did not exist in all accounts, then something is very wrong

Indeed something is wrong, but per the error of mismatches in permissions and internal accounts caught [here](https://github.com/MetaMask/metamask-mobile/issues/16587), we may indeed want to filter... Does feel bad that this implies we need to do this check essentially everywhere we touch account permissions. 

Though as I'm rereading this I'm not sure @wenfix 's intention is to do "a check" he needs to do this to filter down which accounts should be pushed into the defaultAccounts array?

I'm also not sure I follow your preference for appending these accounts outside of `getDefaultAccounts` @jiexi ?

Reply:
> Though as I'm rereading this I'm not sure @wenfix 's intention is to do "a check" he needs to do this to filter down which accounts should be pushed into the defaultAccounts array?

i understand now. This is getting the `MergedInternalAccountWithCaipAccountId` from `CaipAccountId` of `existingPermittedCaipAccountIds`. 

> I'm also not sure I follow your preference for appending these accounts outside of getDefaultAccounts @jiexi

I incorrectly assumed that getDefaultAccounts was returning an array of `CaipAccountId` and so I thought `existingPermittedCaipAccountIds` could trivially be appended to the result of `getDefaultAccounts()` rather than inside of the helper itself. 


Everything I said can be ignored. Sorry about that, @wenfix